com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs

```diff
@@ -2132,6 +2469,143 @@ internal static bool IsBakedShadowMaskLight(Light light)
         }
 
+        static float Log2f(float f)
+        {
+            return Mathf.Log(f, 2); // No way to directly emit the log2() instruction
+        }
+
+        static int CeilLog2i(int i)
+        {
+            return Mathf.CeilToInt(Log2f(i)); // No integer log in our math library
+        }
+
+        static void AssertUnitLength(Vector3 vec)
+        {
+            const float FLT_EPS = 5.960464478e-8f;
+            const float k       = 32; // Empirical constant
+
+            float sqMag = Vector3.SqrMagnitude(vec);
+
+            Debug.Assert(Mathf.Abs(1 - sqMag) < k * FLT_EPS);
+        }
+
+        static Vector3 ComputeWorldSpaceCentroidOfBoundedEntity(Light light)
+        {
+            // Most lights are "centered" by nature.
+            Vector3 centroidWS = light.transform.position;
+
+            HDAdditionalLightData lightData = GetHDAdditionalLightData(light);
+
+            HDLightType  lightType  = lightData.ComputeLightType(light);
+            Debug.Assert(lightType != HDLightType.Directional);
+
+            if (lightType == HDLightType.Spot)
+            {
+                AssertUnitLength(light.transform.forward);
+
+                Vector3 dirWS = light.transform.forward;
+                float   range = lightData.range;
+
+                centroidWS += (0.5f * range) * dirWS;
+            }
+
+            return centroidWS;
+        }
+
+        static Vector3 ComputeWorldSpaceCentroidOfBoundedEntity(HDProbe probe)
+        {
+            return (Vector3)probe.influenceToWorld.GetColumn(3);
+        }
+
+        static float ComputeLinearDepth(Vector3 positionWS, HDCamera hdCamera, int viewIndex, bool forceRWS = false)
+        {
+            Matrix4x4 viewMatrix = GetWorldToViewMatrix(hdCamera, viewIndex); // Non-RWS
+
+            if (forceRWS && (ShaderConfig.s_CameraRelativeRendering != 0))
+            {
+                viewMatrix.SetColumn(3, new Vector4(0, 0, 0, 1));
+            }
+
+            Vector3 positionVS = viewMatrix.MultiplyPoint(positionWS);
+
+            Debug.Assert(viewMatrix.MultiplyVector(hdCamera.camera.transform.forward).z > 0, "The view space z-axis must point forward!");
+
+            return positionVS.z;
+        }
+
+        // 'w' is the linear depth (Z coordinate of the view-space position).
+        // 'f' is the distance to the far plane.
+        // We consider the distance to the near plane n ≈ 0, since that plane may be oblique.
+        static int ComputeFixedPointLogDepth(float w, float f, int numBits = 16)
+        {
+            // z = Log[w/n] / Log[f/n]
+            // Undefined for (w < n, so we must clamp). This should not significantly affect the efficiency of Z-binning.
+            // Still need the distance to the near plane in order for the math to work.
+            // Setting it too low will quickly consume the available bits. 0.1 is a safe value.
+            // See https://zero-radiance.github.io/post/z-buffer/
+            const float n = 0.1f;
```

Comment:
Same question than previously, why are you not using the camera near?

Reply:
Answered above.